src/loaders/amf_load.c

```diff
@@ -169,41 +184,45 @@ static int amf_load(struct module_data *m, HIO_HANDLE *f, const int start)
 		for (i = 0; i < mod->ins; i++) {
 			b = hio_read8(f);
 			if (b != 0 && b != 1) {
-				ver = 0x09;
+				no_loopend = 1;
 				break;
 			}
 			hio_seek(f, 32 + 13, SEEK_CUR);
 			if (hio_read32l(f) > 0x100000) { /* check index */
```

Comment:
index should never be greater than the number of instruments (as per how it's loaded in the AMF loader of DSMI), so this check can be tightened.

Reply:
That's what I thought, I just hadn't looked into it. I've updated this check.